M2/Macaulay2/m2/expressions.m2

```diff
@@ -1156,13 +1156,8 @@ texMath VectorExpression := v -> (
 
 -----------------------------------------------------------------------------
 print =  x -> (
-    c := class x;
-    while not c#?{topLevelMode,print} do (
-	if c === Thing then (<< net x << endl; return); -- default
-	c = parent c;
-	);
-    c#{topLevelMode,print} x
-    )
+    if (f := lookup({topLevelMode, print}, class x)) =!= null then f x
```

Comment:
any reason for not breaking this into two lines?

Reply:
incidentally, wasn't the operator `??` introduced for such situations?

Reply:
Ooh, yeah -- it was!

Reply:
I couldn't come up with a good readable version using `??`, but I did like the idea of moving then `then` statement to its own line.